doc/superbuild/source/developers/testing/Answers.rst

```diff
@@ -0,0 +1,98 @@
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   -----------------------------------------------------------------------------
+
+.. _Answers:
+
+Answer Files
+============
+
+When solving ODEs/DAEs and nonlinear algebraic equations numerically, there is not always a known
+analytical solution for the problem to compare with. Additionally, there may be many paths to a
+correct solution and some may be less efficient than others. This presents an issue for testing
+SUNDIALS. How do we ensure SUNDIALS is working correctly if we cannot write down what "correct" is?
+To solve this problem SUNDIALS has tests (which often also serve as example codes) that have been
+verified by the author to be functioning correctly. These tests produce output which is stored in a
+file known as the "answer" file (aka "out" or "output" file). The output typically consist of the
+solution and/or some quantities derived from the solution as well as statistics about how the
+integration and/or solve (e.g., number of time steps). When changes are made to SUNDIALS, we ensure
+that these output files do not change unless it is expected/desired. Practically, this is ensured
+by our :ref:`CI testing <CI>`.
+
+Due to difference in microarichtectures and the nature of floating point arithmetic, it is quite
+possible that the output generated on one machine may differ from the output generated on another.
+As such, we specify that the the answer files that are embedded in ``examples/`` (the ``.out``
+files) should match what is produced on the Jenkins CI machine.  We also have a `separate git
+repostiory <https://github.com/sundials-codes/answers>`_ which holds answers for other  machines,
+such as the GitHub Actions virtual machines. 
+
+
+Updating Answer Files
+---------------------
+
+Sometimes it is necessary to update the answer files so that the tests pass, if a change in output
+is expected/desired. Changing output files requires careful verification that the output is still
+"correct". The procedure for updating answer files is as follows:
+
+- For the ``.out`` files embedded in examples, simply overwrite the file with the new output. If the
+  tests still fail, then you will need to download the output file generated on the Jenkins CI
+  machine and commit it. Extracting this file from the Jenkins machine may require help from David
+  or Cody currently.
+
+- For the ``.out`` files in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_:
+
+  #. Navigate to `test/answers` in the sundials repository. If there is nothing in this directory,
+     you need to run `git submodule init && git submodule update`.
+
+  #. Checkout the main branch: ``git checkout main``. 
+
+  #. Create a new branch off of main. It is ideal if you use a branch name that matches the name of
+     the branch that you are developing in sundials on: ``git checkout -b <branch name>``.
```

Comment:
Does this work from within the submodule? Looks like the remote is an `https` address so I'd need to setup an [alternative authentication method](https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls). 

Usually I clone the `answers` repo separately push a new branch from there and update the commit in `test/answers`

Reply:
You just have to add a different remote from within the submodule. Ill add some more details about both.